src/vos/ilog.c

```diff
@@ -1080,13 +1080,19 @@ ilog_fetch_move(struct ilog_entries *dest, struct ilog_entries *src)
 }
 
 static void
-ilog_status_refresh(struct ilog_context *lctx, uint32_t intent,
+ilog_status_refresh(struct ilog_context *lctx, uint32_t intent, bool has_cond,
 		    struct ilog_entries *entries)
 {
 	struct ilog_priv	*priv = ilog_ent2priv(entries);
 	struct ilog_entry	 entry;
 	int32_t			 status;
 	bool			 same_intent = (intent == priv->ip_intent);
+	bool			 retry;
+
+	if ((intent == DAOS_INTENT_UPDATE || intent == DAOS_INTENT_PUNCH) && !has_cond)
```

Comment:
can condition to be INTENT as well? as DAOS_COND_INTENT_UPDATE?

Reply:
Yes, that is not impossible, but it seems not simplify the patch. Currently, we handle "intent" and "condition" as two independent parameters, not only in this function, but also some other places. That is how they are transferred from object layer downcall APIs. If not change related APIs, then we need to re-combine them and dispatch them to related functions inside vos. If it is necessary, we can consider that in subsequent patch.